Call applyBlockDefaults from addBlock/insertBlock, add initialValue() configuration option for blocks (#5320)
plone/volto: docs/source/blocks/settings.md

Our new block should be ready to use in the editor.

## Common block options

> It is a common pattern to use the block configuration to allow customization of a block's behavior or to provide block-specific implementation of various Volto mechanisms. Some of these "common" options are:

Reviewer: ```suggestion
It is a common pattern to use the block configuration to allow customization of a block's behavior or to provide block-specific implementation of various Volto mechanisms.
Some of these common options are described in the following sections.
```

### `blockHasValue`

> Returns true if the provided block data represents a value for the current block. Required for alternate default block types implementations. See also [Settings reference](/configuration/settings-reference)

Reviewer: ````suggestion
`blockHasValue` returns `true` if the provided block data represents a value for the current block.
Required for alternate default block types implementations.

```{seealso}
See also [Settings reference](/configuration/settings-reference).
```
````

### `initialValue`

> A function that can be used to get the initial value for a block. It has the

Reviewer: ```suggestion
`initialValue` is a function that can be used to get the initial value for a block.
```

> following signature:

Reviewer: ```suggestion
It has the following signature.
```

> `initialValue({id, value, formData, intl}) => newFormData`

Reviewer: ````suggestion
```jsx
initialValue({id, value, formData, intl}) => newFormData
```
````

### `blockSchema`

> A must-have for modern Volto blocks, this is a function (or directly the schema object) that returns the schema for the block data. While it's not required, having the schema defined enables the block to have its initial value based on the default values declared in the schema.

Reviewer: ```suggestion
A must-have for modern Volto blocks, `blockSchema` is a function, or directly the schema object, that returns the schema for the block data.
Although it's not required, defining the schema enables the block to have its initial value based on the default values declared in the schema.
```
